scalding-core/src/main/scala/com/twitter/scalding/ExecutionOptimizationRules.scala

```diff
@@ -147,10 +151,23 @@ object ExecutionOptimizationRules {
       def apply(tup: (A, B, C, ConcurrentExecutionContext)): Future[(D, E)] =
         (Execution.failFastZip(fn1(tup), fn2(tup))(tup._4))
     }
+
+    case class MapWriteFn[A, B, C, D, E](
```

Comment:
I feel like we already have something like this. I’m on my phone so I can’t find it now easily but I thought we already had a case class function for map + write.

Reply:
Didn't find it too, but I might missed something.

Reply:
https://github.com/twitter/scalding/blob/develop/scalding-core/src/main/scala/com/twitter/scalding/Execution.scala#L742

Looks like it is defined in the MapWrite optimization rule.

Reply:
Done